### DISCUSSION TITLE
part-ii/2006-64/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#74] [Question comments]
Part II 2006 Graph Theory: 1.II .17F 

https://questions.tripos.org/part-ii/2006-64/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

Everything except the last part is bookwork. The bound on a planar graph that has no triangles is a result from lectures about graphs of girth $g$ (and here $g = 4$); the upper bound on the size is $\frac{g}{g-2}(n-2)$, which in this case is equal to $2n - 4$ (the other case to consider when $G$ is acyclic, in which case it has size $n - 1$, but $\forall n \geq 3$, $2n - 4 \geq n - 1$). 

The proof of the last part is similar to the proof of the six-colour theorem from lectures. We can assume $n \geq 3$ and therefore apply the previous result to get that the average degree satisfies
$$\frac{2e(G)}{n} \leq 4 - \frac{8}{n} < 4$$

Therefore $G$ contains a vertex $x$ with $d(x) \leq 3$. We can prove by induction on $n$ that $G$ is 4-colourable, the cases $n \leq 4$ being trivial. Let $x$ have degree less than 4 and consider a 4-colouring of $G - x$ (which exists by the inductive hypothesis). Then we can extend to a 4-colouring of $G$ by colouring $x$ a colour not found in $N(x)$ (at least one of the 4 colours will not be in $N(x)$ since $|N(x)| \leq 3$).

</details>
